hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/operation/HederaCreateOperationTracking.java

```diff
@@ -0,0 +1,137 @@
+        this.dynamicProperties = dynamicProperties;
+    }
+
+    @Override
+    protected boolean failForExistingHollowAccount(MessageFrame frame, Address contractAddress) {
+        if (!dynamicProperties.isLazyCreationEnabled()) {
+            final var hollowAccountID =
+                    matchingHollowAccountId((HederaStackedWorldStateUpdater) frame.getWorldUpdater(), contractAddress);
+
+            return hollowAccountID != null;
+        }
+        return false;
+    }
+
+    @Override
+    protected void sideEffects(MessageFrame frame, MessageFrame childFrame) {
+        // Add an in-progress record so that if everything succeeds, we can externalize the
+        // newly
+        // created contract in the record stream with both its 0.0.X id and its EVM address.
+        // C.f. https://github.com/hashgraph/hedera-services/issues/2807
+        final var updater = (HederaStackedWorldStateUpdater) frame.getWorldUpdater();
+        final var sideEffects = new SideEffectsTracker();
+
+        ContractID createdContractId;
+        final var hollowAccountID = matchingHollowAccountId(updater, childFrame.getContractAddress());
+
+        // if a hollow account exists at the alias address, finalize it to a contract
+        if (hollowAccountID != null) {
+            finalizeHollowAccountIntoContract(hollowAccountID, updater);
+            createdContractId = EntityIdUtils.asContract(hollowAccountID);
+        } else {
+            createdContractId = updater.idOfLastNewAddress();
+        }
+
+        sideEffects.trackNewContract(createdContractId, childFrame.getContractAddress());
+        final var childRecord = creator.createSuccessfulSyntheticRecord(NO_CUSTOM_FEES, sideEffects, EMPTY_MEMO);
+        childRecord.onlyExternalizeIfSuccessful();
+        final var opCustomizer = updater.customizerForPendingCreation();
+        final var syntheticOp = syntheticTxnFactory.contractCreation(opCustomizer);
+        if (dynamicProperties.enabledSidecars().contains(SidecarType.CONTRACT_BYTECODE)) {
+            final var contractBytecodeSidecar = SidecarUtils.createContractBytecodeSidecarFrom(
+                    createdContractId,
+                    childFrame.getCode().getContainerBytes().toArrayUnsafe(),
+                    updater.get(childFrame.getContractAddress()).getCode().toArrayUnsafe());
+            updater.manageInProgressRecord(
+                    recordsHistorian, childRecord, syntheticOp, List.of(contractBytecodeSidecar));
+        } else {
+            updater.manageInProgressRecord(recordsHistorian, childRecord, syntheticOp, Collections.emptyList());
+        }
+    }
+
+    private AccountID matchingHollowAccountId(HederaStackedWorldStateUpdater updater, Address contract) {
+        final var accountID = accountIdFromEvmAddress(updater.aliases().resolveForEvm(contract));
+        final var trackingAccounts = updater.trackingAccounts();
+        if (trackingAccounts.contains(accountID)) {
```

Comment:
For here we probably should extend `AccountAccessor` with `exists()` method passing the `besu Address` type instead of `AccountID`, since inside `hedera-evm` we work with `besu Address`.

Reply:
Not needed, as it was moved to externalizer